### DISCUSSION TITLE
Why does my Kernel said `4.9.292_KernelSU-g0617c4440bfe` although i didn't add "KernelSU" into its name

### DISCUSSION THREAD

**Discussion by th1nhhdk** [tiann/KernelSU#1149] [Q&A] [answered]
- Using `adb shell`, `uname -a` returns `KernelSU` for some reason
```bash
sargo:/ $ uname -a
Linux localhost 4.9.292_KernelSU-g0617c4440bfe #1 repo:android-msm-bonito-4.9-android12L SMP PREEMPT Wed Nov 15 aarch64
```

- If you can help me then thank you very much because i've gone insane trying to figure this out 

- My build commands:
```bash
$ repo init -u https://android.googlesource.com/kernel/manifest -b android-msm-bonito-4.9-android12L
$ repo sync -c -j 40
$ cd private/msm-google
$ curl -LSs "https://raw.githubusercontent.com/tiann/KernelSU/main/kernel/setup.sh" | bash -

$ sed -i -e '/^CONFIG_KPROBES$/d' \
-e '/^CONFIG_HAVE_KPROBES$/d' \
-e '/^CONFIG_KPROBE_EVENTS$/d' \
./arch/arm64/configs/bonito_defconfig

$ cat >> ./arch/arm64/configs/bonito_defconfig << EOF
CONFIG_KPROBES=y
CONFIG_HAVE_KPROBES=y
CONFIG_KPROBE_EVENTS=y
EOF

$ sed -i 's/check_defconfig && //g' ./build.config.bonito_no-cfi
$ cd ../../
$ ./build_bonito.sh
```

- I didn't add anything into `Makefile`
```bash
user@ubuntu:~/android/bonito-kernel$ head private/msm-google/Makefile 
VERSION = 4
PATCHLEVEL = 9
SUBLEVEL = 292
EXTRAVERSION =
NAME = Roaring Lionus
```

**Comment by Coconutat** [comment]
Check your device defconfig.I think they change this:```CONFIG_LOCALVERSION.```

**Reply by th1nhhdk** [reply]
I forgot to say that `grep LOCALVERSION ./arch/arm64/configs/bonito_defconfig` returned nothing.

**Reply by Coconutat** [reply]
I don't know if your kernel is compiled using build scripts. It is very likely that this variable was declared within the compiled script

**Reply by th1nhhdk** [reply]
[build.sh.txt](https://github.com/tiann/KernelSU/files/13387815/build.sh.txt)
- This is android's build script, doesn't seem to affect it.

- It take configuration from here, no mention of `LOCALVERSION`
[build.config.bonito_no-cfi](https://android.googlesource.com/kernel/msm/+/refs/heads/android-msm-bonito-4.9-android12L/build.config.bonito_no-cfi)
[build.config.common.clang](https://android.googlesource.com/kernel/msm/+/refs/heads/android-msm-bonito-4.9-android12L/build.config.common.clang)
[build.config.common](https://android.googlesource.com/kernel/msm/+/refs/heads/android-msm-bonito-4.9-android12L/build.config.common)

**Reply by Coconutat** [reply]
It's really unusual, and I don't quite understand now.

**Comment by th1nhhdk** [comment, marked as answer]
Oh wait, It fixed itself for some reason, I think the problem is that the google build script detects that we are using the kernel from androidsource so it added the `repo:android-msm-bonito-4.9-android12L` part, moving the kernel source to github and build again fixes the problem

**Reply by Coconutat** [reply]
I see, I usually move the original kernel code to Github first and then modify and submit the commit, so I have never encountered this issue.
